src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_tasklet.py

```diff
@@ -296,14 +296,13 @@ def _visit_deref(self, node: gtir.FunCall) -> MemletExpr | ValueExpr:
                     else:
                         assert isinstance(index_expr, SymbolExpr)
 
-                dtype = it.field.desc(self.sdfg).dtype
+                dtype = arg_expr.field.desc(self.sdfg).dtype
                 return self._get_tasklet_result(dtype, deref_node, "val")
 
         else:
-            assert isinstance(it, MemletExpr)
```

Comment:
I would keep this assert, maybe you change something and then forget this place.

Reply:
This was intentional. Actually, this change triggered the whole renaming in this file. The broadcasting of a literal value to a field is implemented as a `deref` node. Therefore, `deref` can return a `SymbolExpr`.